File: CGContextDrawImageScale/CGContextDrawImageScale/ViewController.m
//
//  ViewController.m
//  CGContextDrawImageScale
//
//  Created by gietal-dev on 6/16/17.
//  Copyright © 2017 gietal-dev. All rights reserved.
//

#import "ViewController.h"
#import "ImageView.h"

@implementation ViewController {
    
    __weak IBOutlet NSTextField *scaleTextField;
    __weak IBOutlet ImageView *imageView;
}

- (void)viewDidLoad {
    [super viewDidLoad];


}

-(void)awakeFromNib {
    [super awakeFromNib];
    
    // Do any additional setup after loading the view.
    [scaleTextField setStringValue:@"1"];
    [imageView setChunkSize:CGSizeMake(99, 99)];
}

- (void)setRepresentedObject:(id)representedObject {
    [super setRepresentedObject:representedObject];

    // Update the view, if already loaded.
}

- (IBAction)onScaleSliderChanged:(NSSlider *)sender {
    NSString *str = [NSString stringWithFormat:@"%.2f", [sender floatValue] ];
    [scaleTextField setStringValue:str];
    [imageView setImageScale:[sender floatValue]];
}


@end

File: stripping/stripping/ViewController.swift
//
//  ViewController.swift
//  stripping
//
//  Created by gietal-dev on 9/20/17.
//  Copyright © 2017 gietal-dev. All rights reserved.
//

import Cocoa

public class FindMeClass {
    public func hello() {
        findmeValue = 123
    }
    public var findmeValue = 0
}
class ViewController: NSViewController {
    @IBOutlet weak var label: NSTextField!
    let findmeclass = FindMeClass()
    
    override func viewDidLoad() {
        super.viewDidLoad()

        // Do any additional setup after loading the view.
        findme()
    }

    override var representedObject: Any? {
        didSet {
        // Update the view, if already loaded.
        }
    }
    @IBAction func findmepressed(_ sender: Any) {
        findmeclass.hello()
        label.stringValue = String("findmeValue: \(findmeclass.findmeValue)")
    }

    fileprivate func findme() {
        Swift.print("hello findme")
    }

}

